### PR TITLE
Reconfigure live system

### PR DESCRIPTION
**Pull request by mgax**

### CODE REVIEW

**Review by gabriel-v — CHANGES_REQUESTED** [review]
Please make a few tests that run the agent thing in a few interesting ways, maybe with `thread.sleep` calls and whatnot. Test that you can't run two at the same time. Maybe check for the logs being printed.

**Comment by mgax** [comment]
Yep, you're totally right. Those tests will give us peace of mind.

File: liquidcore/site/settings/common.py [hunk 1]
```diff
@@ -106,3 +106,5 @@
 DISCOVERY_URL = None
 
 INITIALIZE_LOCK_FILE_PATH = str(BASE_DIR / 'initialize.lock')
+
+LIQUID_SETUP_SUDO = False
```
Comment (gabriel-v) [inline liquidcore/site/settings/common.py]: What is this and why do we need a setting here? We either have sudo on the `liquid` user for reconfiguring, or we don't. I think having a global flag for it is crazy. What's the use case here?

File: liquidcore/config/templates/liquidcore/config.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!doctype html>
```
Comment (gabriel-v) [inline liquidcore/config/templates/liquidcore/config.html:1]: And I'm here thinking pretty printed json is luxury

File: liquidcore/config/agent.py [hunk 3]
```diff
@@ -0,0 +1,328 @@
+"""
+Configuration agent for liquid-core
+
+This module does daemonization voodoo. It has two public methods: `launch` and
+`status`.
+
+## `agent.launch(target_configuration, repair)`
+`target_configuration` is the configuration that will be applied.
+
+`repair` - if the last job failed, should we attempt to recover? This flag
```
Comment (gabriel-v) [inline liquidcore/config/agent.py]: Maybe explain what it does. Reruns the last working configuration? Reruns the last broken configuration? Unmarks failures and releases the flock? Mines bitcoin for 2 minutes?

File: liquidcore/config/agent.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+            if psutil.pid_exists(pid):
+                # the job is running, let's wait a bit
+                time.sleep(.2)
+                continue
+
+            # look again to make sure the pidfile was not deleted
+            if self.pid_file.exists():
+                # yep, process is dead but pidfile is in place, it died :(
+                raise JobFailed(
+                    'Job {} died, i found its stale pidfile'
```
Comment (gabriel-v) [inline liquidcore/config/agent.py]: capital I

File: liquidcore/config/agent.py [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    argv = [sys.executable, __file__] + [str(a) for a in args]
+    subprocess.run(argv, check=True, **kwargs)
+
+
+def log(*args):
+    print(*args, timestamp())
+
+
+class State:
+
```
Comment (gabriel-v) [inline liquidcore/config/agent.py]: Extra newline, repeat, extra newline
Cleanup on aisle 206

**Review by gabriel-v — CHANGES_REQUESTED** [review]
Do we save before & after configs? The `repair` flag just clears the lock and reruns the last intent, right?

How about this: if a job fails, then revert to what was before?

File: liquidcore/site/settings/common.py (hunk 1 above)
Comment (mgax) [reply liquidcore/site/settings/common.py]: Mainly so the tests pass without sudo rights. I will change the default to True and change it in `testuite/settings.py`.

**Comment by gabriel-v** [comment]
closes https://github.com/liquidinvestigations/liquidinvestigations/issues/80

**Pull request merged**